src/generator.rs

```diff
@@ -126,33 +123,32 @@ impl Generator {
     /// Execute code generation with pre-configured parameters
     pub async fn generate(&self) -> miette::Result<()> {
         let manifest = Manifest::read().await?;
+        let store = PackageStore::current().into_diagnostic()?;
 
         info!(":: initializing code generator");
 
         ensure!(
-            manifest.package.kind.compilable() || !manifest.dependencies.is_empty(),
+            manifest.package.kind.is_compilable() || !manifest.dependencies.is_empty(),
             "either a compilable package (library or api) or at least one dependency is needed to generate code bindings."
         );
 
         self.run()
             .await
             .wrap_err(miette!("failed to generate bindings"))?;
 
-        if manifest.package.kind.compilable() {
-            let location = Path::new(PackageStore::PROTO_PATH);
+        if manifest.package.kind.is_compilable() {
             info!(
                 ":: compiled {} [{}]",
                 manifest.package.name,
-                location.display()
+                store.proto_path().display()
```

Comment:
Can we rename the methods to something snappier? `store.dir()` and `store.vendor_dir()`? 

I still think that the "proto_path" is not related to the store, so:

proto/* -> the current package
proto/vendor/* -> the "package store"

So if we would split out the logic for accessing the first path that would be great. I think the weirdness that the vendor dir lives inside of the proto dir creates these blurry borders

Reply:
I stuck with the names that were used before: `PROTO_DIR` and `PROTO_VENDOR_DIR`. I'll create an issue for this.